appservice/query/query.go

```diff
@@ -33,7 +34,7 @@ const userIDExistsPath = "/users/"
 
 // AppServiceQueryAPI is an implementation of api.AppServiceQueryAPI
 type AppServiceQueryAPI struct {
-	HTTPClient *http.Client
+	HTTPClient *gomatrixserverlib.Client
```

Comment:
Why are we using this client instead of a normal one? GMSL.Client is for federation traffic only I thought? ASes do not get sent signed traffic.

Reply:
`gomatrixserverlib.Client` is for normal requests (like what the media API uses to fetch media), `gomatrixserverlib.FederationClient` is for signed requests.

Reply:
What's the difference between GMSL.Client and net/http.Client?

Reply:
Not much ,mostly just that `gomatrixserverlib.Client` has all of the correct wiring already to give it a timeout and/or disable TLS validation without needing to reinvent the wheel again in the AS code.